packages/ember-glimmer/lib/components/curly-component.js

```diff
@@ -45,8 +78,8 @@ class CurlyComponentManager {
     let parentView = dynamicScope.view;
 
     let klass = definition.ComponentClass;
-    let attrs = args.named.value();
-    let props = attrsToProps(args.named.keys, attrs);
+    let attrs = processAttrs(args, klass.positionalParams);
+    let props = attrsToProps(keysForAttrs(args, klass.positionalParams), attrs);
```

Comment:
`processAttrs` and `attrsToProps` have basically lost their meaning at this point 😨

Can you extract them into your new file in utils and make it something like `let { attrs, props } = processArgs(args, klass.positionalParams);`?

`attrs` is the thing we pass to the `did*Attrs` hooks, and `props` is what we set on the component instance.

I don't think there is necessarily value of keeping them as separate functions anymore, and passing keys etc. The reason it was factored that way was largely an evolution of the requirements so far, and the new requirements you have introduced here have change the way they work significantly.

I think it could have been much better factored if you weren't subject to these random historical constraints. :smile: